include/votca/xtp/eigencuda.h

```diff
@@ -0,0 +1,116 @@
+/*
+ *            Copyright 2009-2019 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef __XTP_EIGEN_CUDA__H
+#define __XTP_EIGEN_CUDA__H
+
+#include <cublas_v2.h>
+#include <curand.h>
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <vector>
+#include <votca/xtp/eigen.h>
+
+/*
+ * \brief Perform Tensor-matrix multiplications in a GPU
+ *
+ * The `EigenCuda` class handles the allocation and deallocation of arrays on
+ * the GPU.
+ */
+
+namespace votca {
+namespace xtp {
+
+// Unique pointer with custom delete function
+using uniq_double = std::unique_ptr<double, void (*)(double *)>;
```

Comment:
this is in the middle of xtp namespace with a general name, I do not think you need that.

Reply:
fixed